webrender/src/tiling.rs

```diff
@@ -1601,13 +1719,24 @@ impl ScreenTile {
                         // If an opaque primitive covers a tile entirely, we can discard
                         // all primitives underneath it.
                         if layer.xf_rect.as_ref().unwrap().kind == TransformedRectKind::AxisAligned &&
-                           prim_metadata.clip_index.is_none() &&
+                           prim_metadata.clip_cache_info.is_none() &&
                            prim_metadata.is_opaque &&
                            prim_bounding_rect.as_ref().unwrap().contains_rect(&self.rect) {
                             current_task.as_alpha_batch().items.clear();
                         }
                     }
 
+                    // Add a task to render the updated image mask
+                    if let Some(ref clip_info) = prim_metadata.clip_cache_info {
+                        if let Some(mask_task) = RenderTask::new_mask(self.rect, clip_info) {
+                            current_task.children.push(mask_task);
+                        } else {
+                            // The primitive has clip items but their intersection is empty
```

Comment:
Ideally we'd catch this earlier, and not even mark the primitive as visible in this case. But we can do that as a follow up (let's just add a TODO comment here about it).